### DISCUSSION TITLE
Ability to specify custom format for multi-value variables

### DISCUSSION THREAD

**Discussion by Elyahou** [grafana/grafana#45971] [Dashboards, Panels & Visualizations]
Right now, the `raw` format option for variable use comma to separate multi-value variables. I propose to add ability to specify the delimiter and prefix, suffix for such variables.

For example, to use ` OR ` and `(`, `)` as delimiter, prefix and suffix:

`${variable_name:raw:prefix:delimiter:suffix)}`
`${variable_name:raw:(: AND :)}`

It will allow to construct for example sql query parts inside variables:

for example,

This variable:

```
$variable = [' column1 = 1', 'column2 = 1', 'column3 = 2'] 
```

will be able to translate:

```
WHERE ${variable}
```

to 

```
WHERE column1 = 1
```

or with multi-value:

```
WHERE (column1 = 1 OR column2 = 1)
```

**Comment by imageslr** [comment]
I vote for this feature. In my case, I want to add common prefix for each variable so that can be used as OpenTB query tags.

Multi variables:

```
$variable=boo, far
```

I want:
```
*${variable:pipe}* 
*boo*|*far*
```

I got:
```
*${variable:pipe}* 
*boo|far*
```

**Comment by ldrascic** [comment]
⬆️ Upvote! ⬆️

This would be **EPIC** because it would allow dynamic query creation and reusability of dashboard panels. 

Since I am using InfluxDB (Flux language) comma delimiter does not suit my case (I need space delimiter). Here is example in text panel:

![image](https://user-images.githubusercontent.com/124495487/216812824-0af46663-9ca8-405b-baa0-2494a784fda1.png)


I would use this feature to show different OS metrics (CPU usage, Memory, SWAP...) in same table. Currently this is not possible with multiple select dropdown because comma delimiter is automatically added in between variable values.

Environment:
Grafana 9.3.6. OSS
InfluxDB 2.6.1. OSS
Operating System: Fedora Linux 36


Best regards,
ldrascic

**Comment by solarmon** [comment]
I'm currently using Grafana v10.0.1.

This is not quote the same issue/request as the OP for variable type of "custom", but mine is for variable type of "query" (using a Loki datasource), with multi-value selection. This seems to generate a list separate by a vertical pipe ("|"), but my dashboard query using the variable needs/requires the list generated by the variable to be separated by a comma (","). There is no option for the variable query to specify what delimiter character use.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #83014. Feel free to continue the discussion around this there. Thank you!
